src/install-nextstrain.rst

```diff
@@ -0,0 +1,159 @@
+
+`Follow Docker's installation guide <https://docs.docker.com/engine/install/>`_ for your operating system.
+Note that for M1 Mac and Windows users, we recommend sticking with the Conda environment above.
+Docker is not yet ready for widespread use on the M1 Mac.
+Similarly, there are still significant obstacles to running Docker with Windows, as documented in `our issue tracking the problems <https://github.com/nextstrain/cli/issues/31>`_.
+
+After installing and starting Docker, configure the Nextstrain CLI to use Docker as the default environment manager.
+
+.. code-block:: bash
+
+    nextstrain check-setup --set-default
+
+The output of this last command should look like this:
+
+.. code-block:: bash
+
+    Setting default environment to docker.
+
+Finally, download the latest Docker image for Nextstrain.
+
+.. code-block:: bash
+
+    nextstrain update
+
+Next, :doc:`check out the quickstart <tutorials/quickstart>`.
+
+Upgrade an existing installation
+================================
+
+Update the base Conda environment.
+
+.. code-block:: bash
+
+    conda activate base
+    conda update conda
+
+Update the Nextstrain environment.
+
+.. code-block:: bash
+
+    conda activate nextstrain
+    conda update --all
+
+If you are using the Docker image, download the latest version with the Nextstrain CLI.
+
+.. code-block:: bash
+
+    nextstrain update
+
+Troubleshoot a broken installation
+==================================
+
+If Conda fails to install or update Nextstrain using the commands above, it's possible that Conda itself is out-of-date or that Conda cannot figure out how to resolve the environment's dependencies.
+Try the following approaches, to fix these broken installations.
+
+Remove your environment and start from scratch
+----------------------------------------------
+
+Starting from scratch often fixes problems with Conda environments.
+To start over with a new Nextstrain environment, delete your current environment.
+
+.. code-block:: bash
+
+    conda activate base
+    conda env remove -n nextstrain
+
+Then, repeat :ref:`the installation instructions above <install-with-conda>`, starting with the update of Conda itself.
+
+Use Mamba as an alternative to Conda's environment solver
+---------------------------------------------------------
+
+`Mamba <https://github.com/mamba-org/mamba>`_ is a drop-in replacement for most ``conda`` functionality that implements a faster dependency solving algorithm in C++ and multithreaded downloads.
+As a result, Mamba can install Conda packages much faster and more accurately than the original Conda installer.
+
+To try it out, install Mamba.
+
+.. code-block:: bash
+
+    conda install -n base -c conda-forge mamba
+
+Then, use Mamba to create the Nextstrain environment.
+
+.. code-block:: bash
+
+    mamba create -n nextstrain -c conda-forge -c bioconda nextstrain
```

Comment:
I would probably drop mention of Mamba entirely unless we expect its use to be necessary often? People who are familiar with Mamba will already know they can use it without further instruction.

Reply:
I'd like to keep this. I don't see Mamba going away soon (unless the Conda team pulls in its functionality) and it's now the default backend for Snakemake's Conda environment manager (as of [v6.1.0](https://snakemake.readthedocs.io/en/stable/project_info/history.html#id1)). I personally would not have found out about Mamba unless Snakemake mentioned it in the docs, so I would not assume everyone who spends time with Conda knows about it already.

By including its description in the troubleshooting section, we aren't pushing for Mamba as the recommended default (like Snakemake does), but at least we can mention it as a tool used by our team.